### PR TITLE
Some tweaks

### PR DESCRIPTION
**Pull request by smorimoto**
- Add some missing packages

```
jsonrpc.opam: changes needed:
  "base" {with-test & >= v0.14.1}          [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
  "ppx_expect" {with-test & >= v0.14.1}    [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
  "ppx_inline_test" {with-test & >= v0.14.1} [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
lsp.opam: changes needed:
  "result" {>= 1.5}                        [from lsp/src]
  "base" {with-test & >= v0.14.1}          [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
  "ppx_inline_test" {with-test & >= v0.14.1} [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
ocaml-lsp-server.opam: changes needed:
  "base" {with-test & >= v0.14.1}          [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
  "ppx_expect" {with-test & >= v0.14.1}    [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
  "ppx_inline_test" {with-test & >= v0.14.1} [from fiber-unix/test, jsonrpc-fiber/test, lsp-fiber/test]
```

- Update ocamlformat to 0.16.0
- The workflow using opam lock is not that good with the `"jsonrpc" {= version}` part of the lsp package very well
  - You can't update the lock file because version dev can't be locked
  - Because of this, the cache can't be used safely, so use the actions-ml project in the CI environment

### CODE REVIEW

File: .ocamlformat [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version=0.15.1
+version=0.16.0
```
Comment (rgrinberg) [inline .ocamlformat:1]: hold off on this change for now. I prefer to update ocamlformat in vscode, dune, lsp all at once

File: dune-project [hunk 2]
```diff
@@ -28,46 +28,55 @@ Implementation of the LSP protocol in OCaml. It is designed to be as portable as
...
  (synopsis "Jsonrpc protocol implemenation")
  (description "See https://www.jsonrpc.org/specification")
  (depends
-  yojson
-  stdlib-shims
+  (base :with-test)
+  (ocaml (>= 4.06))
+  (odoc :with-doc)
+  (ppx_expect :with-test)
+  (ppx_inline_test :with-test)
```
Comment (rgrinberg) [inline dune-project:77]: This package doesn't even have tests. Why do we have test deps?

File: .ocamlformat (hunk 1 above)
Comment (smorimoto) [reply .ocamlformat:1]: Indeed, I will revert that commit.

File: dune-project (hunk 2 above)
Comment (smorimoto) [reply dune-project:77]: I'm trying to remember some stuff.

**Pull request closed without merging**